src/qofonosimwatcher.cpp

```diff
@@ -58,8 +57,12 @@ void QOfonoSimWatcher::Private::onOfonoAvailableChanged()
         allSims.clear();
         if (!presentSims.isEmpty()) {
             presentSims.clear();
-            Q_EMIT watcher->presentSimListChanged();
-            Q_EMIT watcher->presentSimCountChanged();
+            if (watcher) {
+                Q_EMIT watcher->presentSimListChanged();
+            }
+            if (watcher) {
```

Comment:
Right, yes.  It isn't safe to directly delete the sender of a signal since you don't know if other handlers are still to be called.